### PR TITLE
Mongo db security

### LINKED ISSUE #185: Secure MongoDB

### PR DESCRIPTION
**Pull request by mehtadarshan1**
fixes #185 

and a initialize setting bug

### CODE REVIEW

File: server/settings.js [hunk 1]
```diff
@@ -428,9 +428,26 @@ var updateSettings = function (updateQuery, callback) {
 exports.initialize = function(callback){
     getAllSettings(function(err, allSettingsObj){
         if(err){
-            return callback(err, null);
+            // initialize if does not exist in db
+            if(err.code === 7004){
+                db.resetAllSettings(function(err, result){
+                    if (err){
+                        return callback(err, null);
```
Comment (esteeroramy) [inline server/settings.js]: handle error

File: server/settings.js [hunk 2]
```diff
@@ -428,9 +428,26 @@ var updateSettings = function (updateQuery, callback) {
...
+                    }
+                    getAllSettings(function(err,defaultAllSettings){
+                        if (err){
+                            return callback(err, null);
```
Comment (esteeroramy) [inline server/settings.js]: handle error

File: server/settings.js [hunk 3]
```diff
@@ -428,9 +428,26 @@ var updateSettings = function (updateQuery, callback) {
...
+                        }
+                        allSettings = defaultAllSettings;
+                        return callback(null,'success');
+                    });
+                });
+            } else {
+                return callback(err, null);
```
Comment (Mohammed-JS-Khan) [inline server/settings.js]: what ramy said

**Review by Mohammed-JS-Khan — APPROVED** [review]
lookth goot to meh

**Review by Mohamed-A-Mohamed — APPROVED** [review]

**Review by esteeroramy — APPROVED** [review]

File: server/settings.js [hunk 4]
```diff
@@ -428,9 +428,26 @@ var updateSettings = function (updateQuery, callback) {
 exports.initialize = function(callback){
     getAllSettings(function(err, allSettingsObj){
         if(err){
```
Comment (ahmedqarmout2) [inline server/settings.js:430]: I already made the changes to cover for that

Comment (ahmedqarmout2) [reply server/settings.js:430]: Why are you doing it again

Comment (ahmedqarmout2) [reply server/settings.js:430]: I knew I did that before, someone dropped my work :'( anyway you don't need all of this. In db.js when it fails to find the settings collection just force it to create one. I sent you the code offline

Comment (mehtadarshan1) [reply server/settings.js:430]: done

**Pull request merged**